src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzer.cs

```diff
@@ -0,0 +1,88 @@
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+
+internal sealed partial class NewUnitTestingIncrementalAnalyzerProvider
+{
+    private sealed class NewUnitTestingIncrementalAnalyzer : IUnitTestingIncrementalAnalyzer
+    {
+        private readonly INewUnitTestingIncrementalAnalyzerImplementation _implementation;
+
+        public NewUnitTestingIncrementalAnalyzer(INewUnitTestingIncrementalAnalyzerImplementation implementation)
+            => _implementation = implementation;
+
+        public Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeDocumentAsync(document, bodyOpt, reasons, cancellationToken);
+
+        public Task AnalyzeProjectAsync(Project project, bool semanticsChanged, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeProjectAsync(project, semanticsChanged, reasons, cancellationToken);
+
+        public Task AnalyzeSyntaxAsync(Document document, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeSyntaxAsync(document, reasons, cancellationToken);
+
+        public Task DocumentCloseAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task DocumentOpenAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task DocumentResetAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task ActiveDocumentSwitchedAsync(TextDocument document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task RemoveDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
+        {
+            _implementation.RemoveDocument(documentId);
+            return Task.CompletedTask;
+        }
+
+        public Task RemoveProjectAsync(ProjectId projectId, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentOpenAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentCloseAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task NonSourceDocumentResetAsync(TextDocument textDocument, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public Task AnalyzeNonSourceDocumentAsync(TextDocument textDocument, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => Task.CompletedTask;
+
+        public void LogAnalyzerCountSummary()
```

Comment:
Can this be removed too?

Reply:
yes.  followup PR does taht.  this PR is just the direct port.

Reply:
(Also from API standpoint, perhaps if AnalyzeDocument + AnalyzeProject are being preserved, then we RemoveProject should also be preserved given that RemoveDocument is being preserved. Don't feel too strongly about this though since this code is on life support anyways :))

Reply:
i really don't want to preserve anything that isn't being used.  if we haven't needed it till now, i don't see a good reason to keep it.  and i'd rather investments be made in moving *off* of htis, instead of onto it :)

alternatively, once this works goes in, you could readd it and totally own that piece :)

Reply:
Sounds totally reasonable. I don't think we would add it back as it would probably involve plumbing through layers that we don't understand :) If we really end up needing to change something, I think we would start looking into switching off of the crawler to a pull-based implementation as we discussed.